spec/unit/hanami/application/settings/dotenv_store_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require "hanami/application/settings/dotenv_store"
+require "dotenv"
+
+RSpec.describe Hanami::Application::Settings::DotenvStore do
+  describe "#with_dotenv_loaded" do
+    let(:store) { described_class.new }
+
+    context "dotenv available" do
+      let(:dotenv) { spy(:dotenv) }
+
+      before do
+        allow(store).to receive(:require).and_call_original
+        stub_const "Dotenv", dotenv
+      end
+
+      it "requires and loads a range of dotenv files, specific to the current HANAMI_ENV" do
+        store.with_dotenv_loaded
+
+        expect(store).to have_received(:require).with("dotenv").ordered
+        expect(dotenv).to have_received(:load).ordered.with(
+          ".env.development.local",
+          ".env.local",
+          ".env.development",
+          ".env"
+        )
+      end
+
+      it "returns self" do
+        expect(store.with_dotenv_loaded).to be(store)
+      end
+
+      context "HANAMI_ENV is 'test'" do
+        before do
+          @hanami_env = ENV["HANAMI_ENV"]
+          ENV["HANAMI_ENV"] = "test"
+        end
+
+        after do
+          ENV["HANAMI_ENV"] = @hanami_env
+        end
```

Comment:
this should be an around block with ensure:

```ruby
        around do |example|
          @hanami_env = ENV["HANAMI_ENV"]
          ENV["HANAMI_ENV"] = "test"
          example.run
        ensure
          ENV["HANAMI_ENV"] = @hanami_env
        end
```

Reply:
@solnic with the new implementation where the environment is injected that's no longer needed :slightly_smiling_face: